src/genpy/dynamic.py

```diff
@@ -161,16 +163,26 @@ def generate_dynamic(core_type, msg_cat):
             buff.write(line + '\n')
     full_text = buff.getvalue()
 
-    # Create a temporary directory
-    tmp_dir = tempfile.mkdtemp(prefix='genpy_')
-
-    # Afterwards, we are going to remove the directory so that the .pyc file gets cleaned up if it's still around
-    atexit.register(shutil.rmtree, tmp_dir)
-
-    # write the entire text to a file and import it (it will get deleted when tmp_dir goes - above)
-    tmp_file = tempfile.NamedTemporaryFile(mode='w', suffix='.py', dir=tmp_dir, delete=False)
-    tmp_file.file.write(full_text)
-    tmp_file.file.close()
+    if tmp_dir_path:
+        # Create a temporary directory with specified path
+        tmp_dir = os.mkdir(tmp_dir_path)
```

Comment:
This should probably handle the case that the directory already exists.

Reply:
My thought process was that if an `output_file` was given, it would overwrite the file if it exists. However, if the file does not exist, it would create a file at the given `output_file` path.

Reply:
By just removing this line it doesn't handle the case anymore where the directory doesn't exist.

Reply:
> By just removing this line it doesn't handle the case anymore where the directory doesn't exist.

@jimwelagu This comment is still pending.

Reply:
@jimwelagu Friendly ping.